addons/base_automation/models/base_automation.py

```diff
@@ -318,6 +320,8 @@ def make_write():
             def write(self, vals, **kw):
                 # retrieve the action rules to possibly execute
                 actions = self.env['base.automation']._get_actions(self, ['on_write', 'on_create_or_write'])
+                if not (actions and self):
```

Comment:
IMHO, it would be clearer:  

`if not actions or not self`

Reply:
I just noticed you wrote this line following [this comment](https://github.com/odoo/odoo/pull/52134#discussion_r432267271), so forget it